AlphaWallet/Settings/Coordinators/CheckTransactionStateCoordinator.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  CheckTransactionStateCoordinator.swift
+//  AlphaWallet
+//
+//  Created by Vladyslav Shepitko on 07.03.2022.
+//
+
+import Foundation
+import PromiseKit
+import web3swift
+
+protocol CheckTransactionStateCoordinatorDelegate: class {
+    func didComplete(coordinator: CheckTransactionStateCoordinator)
+}
+
+class CheckTransactionStateCoordinator: Coordinator {
+    private let navigationController: UINavigationController
+    private let config: Config
+    private var serverSelection: ServerSelection = .server(server: .server(.main))
+    private lazy var rootViewController: CheckTransactionStateViewController = {
+        let viewModel = CheckTransactionStateViewModel(serverSelection: serverSelection)
+        let viewController = CheckTransactionStateViewController(viewModel: viewModel)
+        viewController.configure(viewModel: viewModel)
+        viewController._delegate = self
+
+        return viewController
+    }()
+    private weak var notificationAlertController: UIViewController?
+
+    var coordinators: [Coordinator] = []
+    weak var delegate: CheckTransactionStateCoordinatorDelegate?
+
+    init(navigationController: UINavigationController, config: Config) {
+        self.navigationController = navigationController
+        self.config = config
+    }
+
+    func start() {
+        navigationController.present(rootViewController, animated: false)
+    }
+
+    private var presentationViewController: UIViewController {
+        guard let keyWindow = UIApplication.shared.firstKeyWindow else { return navigationController }
+
+        if let controller = keyWindow.rootViewController?.presentedViewController {
+            return controller
+        } else {
+            return navigationController
+        }
+    }
+
+    private func displayErrorMessage(_ message: String, title: String? = .none) {
+        if let presentedController = notificationAlertController {
+            presentedController.dismiss(animated: true) { [weak self] in
+                guard let strongSelf = self else { return }
+                strongSelf.notificationAlertController = strongSelf.presentationViewController.displaySuccess(title: title, message: message)
+            }
+        } else {
+            notificationAlertController = presentationViewController.displaySuccess(title: title, message: message)
+        }
+    }
+}
+
+extension CheckTransactionStateCoordinator: SelectTransactionHashViewControllerDelegate {
+
+    func didSelectedCheckTransactionStatus(in viewController: CheckTransactionStateViewController, transactionHash: String) {
+        guard let server = serverSelection.asServersArray.first else { return }
+        guard let web3 = try? getCachedWeb3(forServer: server, timeout: 6) else { return }
```

Comment:
Can we move more of this logic to a non-UI logic object? Like `AddCustomChain`. Then keep the coordinator more as glue for UIViewControllers (like storyboards are used for navigation), and simpler.